include/picongpu/fields/incidentField/profiles/Free.hpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2022 Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include "picongpu/fields/incidentField/profiles/Free.def"
+
+
+namespace picongpu
+{
+    namespace fields
+    {
+        namespace incidentField
+        {
+            namespace profiles
+            {
+                template<typename T_FunctorIncidentE, typename T_FunctorIncidentB>

```

Comment:
Should we by default set `T_FunctorIncidentB` to `ZeroFunctor`?

Reply:
No, since zero-functor is not a good default value. However I have an idea for a follow-up PR to introduce functor that calculates B from E using a plane-wave approximation (we'll need this functor internally anyways, so why not also offer to a user) and so use this new `BFromE<T_FunctorIncidentE>` as a default second argument.